AnnoDesigner/ViewModels/UpdateSettingsViewModel.cs

```diff
@@ -0,0 +1,309 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Input;
+using AnnoDesigner.Core.Models;
+using AnnoDesigner.Models;
+using NLog;
+using MessageBox = Xceed.Wpf.Toolkit.MessageBox;
+
+namespace AnnoDesigner.ViewModels
+{
+    public class UpdateSettingsViewModel : Notify
+    {
+        private static readonly Logger logger = LogManager.GetCurrentClassLogger();
+
+        private readonly ICommons _commons;
+        private readonly IAppSettings _appSettings;
+
+        private bool _automaticUpdateCheck;
+        private bool _updateSupportsPrerelease;
+        private string _versionValue;
+        private string _fileVersionValue;
+        private string _presetsVersionValue;
+        private bool _isUpdateAvailable;
+        private bool _isPresetUpdateAvailable;
+        private bool _isAppUpToDate;
+        private bool _isUpdateError;
+        private bool _isBusy;
+        private string _busyContent;
+
+        public UpdateSettingsViewModel(ICommons commonsToUse,
+            IAppSettings appSettingsToUse)
+        {
+            _commons = commonsToUse;
+            _appSettings = appSettingsToUse;
+
+            CheckForUpdatesCommand = new RelayCommand(ExecuteCheckForUpdates);
+            OpenReleasesCommand = new RelayCommand(ExecuteOpenReleases);
+            DownloadPresetsCommand = new RelayCommand(ExecuteDownloadPresets);
+        }
+
+        private AvailableRelease FoundPresetRelease { get; set; }
+
+        public async Task CheckForUpdates(bool isAutomaticUpdateCheck)
+        {
+            try
+            {
+                IsUpdateError = false;
+                IsAppUpToDate = false;
+                IsUpdateAvailable = false;
+                IsPresetUpdateAvailable = false;
+
+                BusyContent = Localization.Localization.Translations["UpdatePrefBusyCheckUpdates"];
+                IsBusy = true;
+
+                await CheckForNewAppVersionAsync(isAutomaticUpdateCheck);
+
+                await CheckForPresetsAsync(isAutomaticUpdateCheck);
+
+                if (!isAutomaticUpdateCheck)
+                {
+                    IsAppUpToDate = !IsUpdateAvailable && !IsPresetUpdateAvailable;
+                }
+
+                IsBusy = false;
+            }
+            catch (Exception ex)
+            {
+                logger.Error(ex, "Error checking version.");
+
+                IsUpdateError = true;
+
+                IsBusy = false;
+
+                if (isAutomaticUpdateCheck)
+                {
+                    MessageBox.Show(Application.Current.MainWindow,
+                        $"Error checking version.{Environment.NewLine}{Environment.NewLine}More information is found in the log.",
```

Comment:
Do we want to localize all these message boxes, while we're in the process of refactoring this code? This would go towards fixing #71 (localize message boxes).

Reply:
I think this is out of scope for this PR.
Best would be a service/helper as abstraction for MessageBoxes. But there is already an issue for this.